src/hdx_cli/cli_interface/migrate/resource_adapter.py

```diff
@@ -0,0 +1,306 @@
+                                     resource_settings: dict | None,
+                                     parent_path=""
+                                     ) -> dict:
+    def is_empty(value):
+        return value in (None, {}, [], "")
+
+    global first_time_user_input
+    adapted_resource_settings = {}
+
+    for field_name, field_props in resource_structure.items():
+        if field_props.get("read_only", False):
+            continue
+
+        is_required = field_props.get("required", False)
+        field_type = field_props.get("type", "")
+        resource_settings_value = resource_settings.get(field_name) if resource_settings else None
+
+        current_path = f"{parent_path}.{field_name}" if parent_path else field_name
+
+        if field_type == "nested object":
+            children_structure = field_props.get("children", {})
+            child_structure = field_props.get("child", None)
+
+            if children_structure:
+                if not resource_settings_value and not is_required:
+                    continue
+                nested_dict = _adapt_resource_to_api_structure(
+                    children_structure,
+                    resource_settings_value or {},
+                    parent_path=current_path
+                )
+                if nested_dict or is_required:
+                    adapted_resource_settings[field_name] = nested_dict
+
+            elif child_structure:
+                if resource_settings_value and isinstance(resource_settings_value, dict):
+                    adapted_resource_settings[field_name] = {}
+                    for key, value in resource_settings_value.items():
+                        adapted_resource_settings[field_name][key] = value
+
+        else:
+            if resource_settings_value is not None and not is_empty(resource_settings_value):
+                adapted_resource_settings[field_name] = resource_settings_value
+            elif is_required:
+                if first_time_user_input:
+                    logger.info("In progress")
+                    logger.info(f"{MISSING_VALUES_HEADER:*^40}")
+                    logger.info(f"* The following fields are required to proceed:")
+                    first_time_user_input = False
+
+                new_value = get_user_value_input(current_path, field_type)
+                adapted_resource_settings[field_name] = new_value
+            else:
+                logger.debug(f"Field '{current_path}' was omitted because it is empty or None.")
+
+    return adapted_resource_settings
+
+
+def adapt_resource_to_api_structure(profile: ProfileUserContext,
+                                    resource_url: str,
+                                    resource_settings: dict
+                                    ) -> dict:
+    resource_structure = get_resource_settings_structure(profile, resource_url)
+    if not resource_structure:
+        return resource_settings
+
+    global first_time_user_input
```

Comment:
you defined this in the _adapt_resource_to_api_structure() too. this might create collisions

Reply:
Yes, got it. Thanks!
I fixed it in the last commit. Please take a look